src/BridgeFactory.sol

```diff
@@ -0,0 +1,117 @@
+        );
+        _cachedThis = address(this);
+    }
+
+    modifier onlyWallet() {
+        require(address(this).code.length == 23);
+        _;
+    }
+
+    modifier noDelegateCall() {
+        require(address(this) == _cachedThis);
+        _;
+    }
+
+    function isValidSignature(
+        bytes32 _hash,
+        bytes calldata _signature
+    ) external view override returns (bytes4) {
+        bytes32[] calldata proof;
+        assembly ("memory-safe") {
+            // _signature is just going to be the proof, then we can read it as so
+            proof.offset := add(0x20, _signature.offset)
+            proof.length := calldataload(proof.offset)
+        }
+        bytes32 root = MerkleProofLib.verify(proof, _hash);
+
+        assembly ("memory-safe") {
+            let ptr := mload(0x40)
+
+            mstore(0x00, 0x60265f8160095f39f35f5f365f5f37365f6c)
+            extcodecopy(address(), 0x20, 0x09, 0x1d)
+            let initCodeHash := keccak256(0x0e, 0x2f)
+
+            mstore(0x00, 0xff00000000000000) // 0xff with padding for factory address
+            mstore(0x2d, root) // salt
+            mstore(0x4d, initCodeHash) // initCode hash
+            let computedAddress := and(
+                keccak256(0x18, 0x55),
+                0xffffffffffffffffffffffffffffffffffffffff
+            )
+
+            // restore clobbered slots
+            mstore(0x60, 0x00)
+            mstore(0x40, ptr)
+
+            if xor(computedAddress, address()) {
+                return(0x00, 0x00)
+            }
+            mstore(0x00, 0x1626ba7e)
+            return(0x1c, 0x04)
+        }
+    }
+
+    function deploy(
+        bytes32 salt
+    ) external noDelegateCall returns (address proxy) {
+        assembly ("memory-safe") {
+            // create a minimal proxy targeting this contract
+            mstore(0x1d, 0x5af43d5f5f3e6022573d5ffd5b3d5ff3)
+            mstore(0x0d, address())
+            mstore(0x00, 0x60265f8160095f39f35f5f365f5f37365f6c)
+            proxy := create2(0x00, 0x0e, 0x2f, salt)
+            if iszero(proxy) {
+                mstore(0x00, 0x30116425) // selector for `DeploymentFailed()`.
+                revert(0x1c, 0x04)
+            }
+        }
+    }
+
+    function approvePermit2(IERC20 token) external onlyWallet returns (bool) {
+        token.safeApprove(
+            0x000000000022D473030F116dDEE9F6B43aC78BA3,
+            type(uint256).max
+        );
+        return true;
+    }
+
+    function call() external {}
+
+    function cleanup() external {}
+
+    function _msgSender() internal view virtual override returns (address) {
+        return msg.sender;
+    }
+
+    function _msgData()
+        internal
+        view
+        virtual
+        override
+        returns (bytes calldata)
+    {
+        return msg.data;
+    }
+
+    function _isForwarded() internal view virtual override returns (bool) {
+        return false;
+    }
```

Comment:
Do we need this here, or can we inherit from `Context`?

Reply:
yeah, surely not needed, added it just to push